publication-commons/src/main/java/no/unit/nva/publication/UserAccessValidationUtil.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.publication;
+
+import static java.util.Objects.nonNull;
+import static no.unit.nva.model.role.Role.CREATOR;
+import static nva.commons.apigateway.AccessRight.EDIT_ALL_NON_DEGREE_RESOURCES;
+import static nva.commons.apigateway.AccessRight.EDIT_OWN_INSTITUTION_RESOURCES;
+import static nva.commons.apigateway.AccessRight.PUBLISH_DEGREE;
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationStatus;
+import no.unit.nva.model.Reference;
+import no.unit.nva.model.instancetypes.PublicationInstance;
+import no.unit.nva.model.instancetypes.degree.DegreeBachelor;
+import no.unit.nva.model.instancetypes.degree.DegreeMaster;
+import no.unit.nva.model.instancetypes.degree.DegreePhd;
+import no.unit.nva.model.pages.Pages;
+import no.unit.nva.publication.model.business.UserInstance;
+import nva.commons.apigateway.AccessRight;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.UnauthorizedException;
+
+public class UserAccessValidationUtil {
```

Comment:
almost everything here uses/is bades on Publication publication and RequestInfo requestInfo, maybe have a constructor for those? also maybe add Publication to the name so to show that this class is dedicated to publication

Reply:
I might have been too general when implementing this. The idea was to create a common validator class to handle all validation, not just deletion of published publications. Therefore the method itself was signed with "...deletingPublishedPublication()". But I can agree one shouldn't add unnecessary complexion for stuff that might or might not be implemented in the future.

Reply:
Making a constructor, as you say, might be a better solution. Will look into it.